sidecarDockerrun.sh

```diff
@@ -13,32 +18,76 @@ cleanup() {
   exit 0
 }
 
-sed -i "s/LogFilePath: \"fuse_log.txt\"/LogFilePath: \"\/data\/manifest-sync-status.log\"/g" ~/fuse-config.yaml
+sed -i "s/LogFilePath: \"fuse_log.txt\"/LogFilePath: \"\/data\/_manifest-sync-status.log\"/g" ~/fuse-config.yaml
 trap cleanup SIGTERM
+
+declare -A TOKEN_JSON  # requires Bash 4
+TOKEN_JSON['default']=`curl http://workspace-token-service.$NAMESPACE/token/?idp=default 2>/dev/null | jq -r '.token'`
+
 while true; do
-    if [ $(df /data/man* | sed '1d' | wc -l) -lt 7 ]; then # remove header line, and also consider the existence of manifest snyc log
-        resp=`curl http://manifestservice-service.$NAMESPACE/ -H "Authorization: bearer $TOKEN_JSON" 2>/dev/null`
+
+    EXTERNAL_OIDC=`curl http://workspace-token-service.$NAMESPACE/external_oidc/ -H "Authorization: bearer ${TOKEN_JSON['default']}" 2>/dev/null | jq -r '.providers'`
+
+    # list of IDPs to get manifests from.
+    # only select IDPs the user is logged into
+    IDPS=( "default" )
+    BASE_URLS=( "https://$HOSTNAME" )
+    for row in $(echo "${EXTERNAL_OIDC}" | jq -r '.[] | @base64'); do
+        _jq() {
+            echo ${row} | base64 -d | jq -r ${1}
+        }
+        # if user is connected, add IDP info to the lists
+        if [ `_jq '.refresh_token_expiration'` != "null" ]; then
+            IDPS+=( `_jq '.idp'` )
+            BASE_URLS+=( `_jq '.base_url'` )
+        fi
+    done
+
+    for i in "${!IDPS[@]}"; do
```

Comment:
What is the `!` for?

Reply:
`${L[@]}` iterates over the list values and `${!L[@]}` iterates over indices